source/dub/dub.d

```diff
@@ -277,28 +277,104 @@ class Dub {
 
 		The script above can be invoked with "dub --single test.d".
 	*/
+		struct Parser {
+			static typeof(this) opCall(string sc) {
+				Parser ret;
+				ret.sourcecode = sc;
+				ret.popFront;
+				return ret;
+			}
+			string sourcecode;
+			string front;
+			bool empty;
+			char inComment = '\0';
+			bool hadFileName;
+			bool hadData;
+			int depth;
+			void popFront() {
+				import std.uni : isWhite;
+				if (empty) return;
+				for (;sourcecode.length > 0;) {
+					if (sourcecode[0].isWhite) {
+						sourcecode = sourcecode[1..$];
+					} else {
+						break;
+					}
+				}
+				if (sourcecode.length <= 0) {
+					empty = true;
+					return;
+				}
+
+				int i,j;
+				for (; i<sourcecode.length; i++) {
+					if (sourcecode[i] == '/') {
+						j=i;
+						if (inComment=='\0') {
+							j = i;
+							i++;
+							if (sourcecode[i] == '+' || sourcecode[i]=='*') {
+								depth++;
```

Comment:
On Thu, Jun 9, 2016 at 8:15 AM, Sönke Ludwig <notifications@github.com>
wrote:

> In source/dub/dub.d
> <https://github.com/dlang/dub/pull/872#discussion_r66387841>:
>
> > +					}
> > +				}
> > +				if (sourcecode.length <= 0) {
> > +					empty = true;
> > +					return;
> > +				}
> > +
> > +				int i,j;
> > +				for (; i<sourcecode.length; i++) {
> > +					if (sourcecode[i] == '/') {
> > +						j=i;
> > +						if (inComment=='\0') {
> > +							j = i;
> > +							i++;
> > +							if (sourcecode[i] == '+' || sourcecode[i]=='*') {
> > +								depth++;
>
> Only /+ +/ supports nesting. /* /* */ */ would be parsed as /* /* */, *, /
> by the compiler.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dlang/dub/pull/872/files/6d6ee0e79d1f8a8ab1bc2beff4ea760963d618d4#r66387841>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABU8Ca5R1Xh308M9nUskZ6LL2lVMDbrqks5qJ69xgaJpZM4IxbiW>
> .
>
At the moment this will just let the compiler show the error to the user.